lib/src/barback/dart2js_transformer.dart

```diff
@@ -395,7 +396,13 @@ class _BarbackCompilerProvider implements dart.CompilerProvider {
     return new Future.sync(() {
       // Find the corresponding asset in barback.
       var id = _sourceUrlToId(url);
-      if (id != null) return _transform.readInputAsString(id);
+      if (id != null) {
+        if (id.extension == '.dill') {
+          return _transform.readInput(id);
```

Comment:
`_readResource()` is declared to return a `Future<String>`, but `readInput()` returns a `Stream<List<int>>`.

Reply:
Done. Dart2js expects both String or List<int>, however I don't understand how returning a Stream didn't cause a crash for me before :-/. Anyhow, I just re-tested this on the SDK repo and it appears to work as expected.